src/main/java/fr/jmmc/oiexplorer/core/gui/chart/RulerOverlay.java

```diff
@@ -0,0 +1,262 @@
+        final Point2D pointOrigin = this.chartPanel.translateScreenToJava2D(p);
+        XYPlot plot = this.chartPanel.getChart().getXYPlot();
+
+        Rectangle2D dataArea = null;
+
+        final PlotRenderingInfo plotInfo = chartPanel.getChartRenderingInfo().getPlotInfo();
+
+        final Rectangle2D plotDataArea = plotInfo.getDataArea();
+
+        if (plotDataArea.contains(pointOrigin)) {
+            dataArea = plotDataArea;
+        } else {
+            dataArea = plotInfo.getDataArea();
+        }
+
+        if (dataArea != null) {
+            final ValueAxis xAxis = plot.getDomainAxis();
+            final ValueAxis yAxis = plot.getRangeAxis();
+
+            final RectangleEdge xAxisEdge = plot.getDomainAxisEdge();
+            final RectangleEdge yAxisEdge = plot.getRangeAxisEdge();
+
+            switch (translation) {
+                case VALUES_TO_COORDS:
+                    p = new PointDouble(xAxis.valueToJava2D(p.getX(), dataArea, xAxisEdge),
+                            yAxis.valueToJava2D(p.getY(), dataArea, yAxisEdge));
+                    break;
+                case COORDS_TO_VALUES:
+                    p = new PointDouble(xAxis.java2DToValue(p.getX(), dataArea, xAxisEdge),
+                            yAxis.java2DToValue(p.getY(), dataArea, yAxisEdge));
+                    break;
+            }
+        }
+        return p;
+    }
+
+    /**
+     * return the distance between the two points of the measure
+     *
+     * @return
+     */
+    private double calculateMeasure() {
+        return Math.sqrt(Math.pow((end.getX() - origin.getX()), 2) + Math.pow((end.getY() - origin.getY()), 2));
+    }
+
+    private double calculateAngle() {
+        return Math.toDegrees(Math.atan2((end.getX() - origin.getX()), (end.getY() - origin.getY())));
+    }
+
+    @Override
+    public void paintOverlay(Graphics2D g2, ChartPanel chartPanel) {
+        if (this.rulerState != RulerState.DISABLED) {
+            g2.clip(chartPanel.getChartRenderingInfo().getPlotInfo().getDataArea());
+            g2.setColor(Color.green);
+            g2.setStroke(new BasicStroke(2));
+
+            PointDouble originCoords = translate(origin, Translation.VALUES_TO_COORDS);
+            PointDouble endCoords = translate(end, Translation.VALUES_TO_COORDS);
+            g2.drawRect((int) originCoords.getX() - RECT_SIZE, (int) originCoords.getY() - RECT_SIZE, RECT_SIZE * 2, RECT_SIZE * 2);
+            g2.drawRect((int) endCoords.getX() - RECT_SIZE, (int) endCoords.getY() - RECT_SIZE, RECT_SIZE * 2, RECT_SIZE * 2);
+            g2.drawLine((int) originCoords.getX(), (int) originCoords.getY(), (int) endCoords.getX(), (int) endCoords.getY());
+
+            g2.setClip(chartPanel.getChartRenderingInfo().getChartArea());
+            g2.setColor(Color.black);
+            g2.drawString("Point 1: x=" + String.format("%.5f", origin.getX()) + " y=" + String.format("%.5f", origin.getY()), 100, chartPanel.getSize().height - 120);
```

Comment:
Use 1 single String.format() call like instead of ++:
String.format("Point 1: x= %.5f  y= %.5f, ...)

Reply:
Factorize chartPanel.getSize().height for better readability